### PR TITLE
[Autodiscover] Handle input-not-finished errors in config reload

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds special error handling in config reload flow to handle state errors specially by not logging in ERROR level to avoid confusion in logging.

Current state:
```
2020-08-26T07:05:05.386Z        ERROR   [autodiscover]  cfgfile/list.go:95      Error creating runner from config: Can only start an input when all related states are finished: {Id: ea745ab688be85a9-native::1308836-2049, Finished: false, Fileinfo: &{628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b-json.log 99 416 {29223485 63734022237 0x608b880} {2049 1308836 1 33184 0 0 0 0 99 4096 8 {1598425436 865214956} {1598425437 29223485} {1598425437 29223485} [0 0 0]}}, Source: /var/lib/docker/containers/628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b/628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b-json.log, Offset: 42864, Timestamp: 2020-08-26 07:05:03.816329242 +0000 UTC m=+187.623544858, TTL: -1ns, Type: container, Meta: map[stream:stdout], FileStateOS: 1308836-2049}
```

Patch:
```
2020-09-02T08:19:12.330Z	DEBUG	[autodiscover]	cfgfile/list.go:97	Error creating runner from config: Can only start an input when all related states are finished: {Id: native::5784858-64769, Finished: false, Fileinfo: &{mytarget2_default_test-63b37f8e982045d4bdac75d302e687e788e979ffeb857b5c1e754f019275d6e5.log 15108 416 {330272804 63734631529 0x36610e0} {64769 5784858 1 33184 0 0 0 0 15108 4096 32 {1599034103 795344632} {1599034729 330272804} {1599034729 330272804} [0 0 0]}}, Source: /var/log/containers/mytarget2_default_test-63b37f8e982045d4bdac75d302e687e788e979ffeb857b5c1e754f019275d6e5.log, Offset: 15465, Timestamp: 2020-09-02 08:19:05.456366051 +0000 UTC m=+12.261964777, TTL: -1ns, Type: container, Meta: map[], FileStateOS: 5784858-64769}
```

## Why is it important?
To avoid confusion in logging because of a non actual ERROR is logged as an ERROR. We need to log with lower severity this kind of errors.

## How to test this PR locally

Using testing steps from https://github.com/elastic/beats/pull/20305.

## Related issues

- Closes https://github.com/elastic/beats/issues/20850
- Relates https://github.com/elastic/beats/pull/20305, https://github.com/elastic/beats/issues/11834, https://github.com/elastic/beats/pull/20898#pullrequestreview-480549739

### CODE REVIEW

File: libbeat/common/errors.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package input
+package common
```
Comment (jsoriano) [inline libbeat/common/errors.go:18]: Why moving this to common? I think it is ok if we have to explicitly import this error for the input package.

**Review by jsoriano — APPROVED** [review]
Looks good, thanks for fixing this. Only a question that can be ignored if you have good reasons :slightly_smiling_face:

File: libbeat/common/errors.go (hunk 1 above)
Comment (ChrsMark) [reply libbeat/common/errors.go:18]: I thought it won't be a good idea to import a `filebeat` package (input) within a `libbeat` package. To my mind `libbeat` is superior from various beats' codebases. Nevertheless I gave it a try and it seems that it leads to a circular import indeed 🙂 :
```
import cycle not allowed
package github.com/elastic/beats/v7/filebeat
	imports github.com/elastic/beats/v7/filebeat/cmd
	imports github.com/elastic/beats/v7/filebeat/beater
	imports github.com/elastic/beats/v7/filebeat/autodiscover/builder/hints
	imports github.com/elastic/beats/v7/filebeat/fileset
	imports github.com/elastic/beats/v7/libbeat/cfgfile
	imports github.com/elastic/beats/v7/filebeat/input
	imports github.com/elastic/beats/v7/filebeat/channel
	imports github.com/elastic/beats/v7/libbeat/cfgfile
```

Comment (jsoriano) [reply libbeat/common/errors.go:18]: Oh, I see. Ok, let's go with this by now.

**Review by jsoriano — APPROVED** [review]

**Comment by thernstig** [comment]
@ChrsMark I am confused by this change. I just ended up with this problem:

```sh
2020-09-24T13:04:39.398Z        ERROR   [autodiscover]  cfgfile/list.go:96      Error creating runner from config: Can only start an input when all related states are finished: {Id:2069814-64771 Finished:false Fileinfo:0xc42018c340 Source:/var/log/containers/serverr-57cd7746ff-pzzlg_default_server-e48194c34dcac07db908e93c44ae34d5130a28ae79766b7645ff3686ff66ef4d.log Offset:12941 Timestamp:2020-09-24 13:04:38.095112478 +0000 UTC m=+1743.843569158 TTL:-1ns Type:log Meta:map[] FileStateOS:2069814-64771}
```

I have no idea what this means, but at least I glad that as a user I could run `kubectl logs <name>` to check the logs of Filebeat, to further google the error.

1. Does this change now hide this error?
2. I still have no idea why this error happens, and googling it the info is sparse.

**Comment by ChrsMark** [comment]
> @ChrsMark I am confused by this change. I just ended up with this problem:
> 
> ```shell
> 2020-09-24T13:04:39.398Z        ERROR   [autodiscover]  cfgfile/list.go:96      Error creating runner from config: Can only start an input when all related states are finished: {Id:2069814-64771 Finished:false Fileinfo:0xc42018c340 Source:/var/log/containers/serverr-57cd7746ff-pzzlg_default_server-e48194c34dcac07db908e93c44ae34d5130a28ae79766b7645ff3686ff66ef4d.log Offset:12941 Timestamp:2020-09-24 13:04:38.095112478 +0000 UTC m=+1743.843569158 TTL:-1ns Type:log Meta:map[] FileStateOS:2069814-64771}
> ```
> 
> I have no idea what this means, but at least I glad that as a user I could run `kubectl logs <name>` to check the logs of Filebeat, to further google the error.
> 
> 1. Does this change now hide this error?
> 2. I still have no idea why this error happens, and googling it the info is sparse.

Hi!

First of all this "error" is sth that has to do with the state of files that are being harvested. It should go away once the harvesting is completed and Autodiscover will be able to start the input again. So imagine that this error could be ignored. This is the reason we turned this error from ERROR level to DEBUG level. This change will be landed with `7.10` release.

**Comment by thernstig** [comment]
@ChrsMark Hi! Ok, I must have misunderstood this thread https://github.com/elastic/beats/issues/11834. There it seems to state that the only way for the logs to be picked up is to restart Filebeat.

**Pull request merged**